common/strings/mem-block.h

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef COMMON_STRINGS_MEM_BLOCK_H
+#define COMMON_STRINGS_MEM_BLOCK_H
+
+#include <string>
+
+#include "absl/strings/string_view.h"
+
+namespace verible {
+// A representation of a block of memory that is owned by MemBlock.
+// Recommended use is to create it somewhere and pass a pointer in a
+// std::unique_ptr<> to explicitly pass unique ownership in a move semantic.
+class MemBlock {
+ public:
+  virtual ~MemBlock() {}
+  virtual absl::string_view AsStringView() const = 0;
+
+ protected:
+  MemBlock() = default;
+
+ private:
+  MemBlock(const MemBlock &) = delete;
+  MemBlock(MemBlock &&) = delete;
```

Comment:
Recommend deleting the default assignment operators as well.

Reply:
Sounds good, done.